contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -0,0 +1,860 @@
+        require(_asset == WETH, "Unsupported asset");
+        require(_amount > 0, "Must deposit something");
+        emit Deposit(_asset, address(0), _amount);
+    }
+
+    /**
+     * @notice Rebalance the pool to the desired token split and Deposit any WETH on the contract to the
+     * underlying aerodrome pool. Print the required amount of corresponding OETHb. After the rebalancing is 
+     * done burn any potentially remaining OETHb tokens still on the strategy contract.
+     * 
+     * Exact _amountToSwap, _minTokenReceived & _swapWETH parameters shall be determined by simulating the 
+     * transaction off-chain. The strategy checks that after the swap the share of the tokens is in the 
+     * expected ranges.
+     * 
+     * @param _amountToSwap The amount of the token to swap
+     * @param _minTokenReceived Slippage check -> minimum amount of token expected in return
+     * @param _swapWETH Swap using WETH when true, use OETHb when false
+     */
+    function rebalance(uint256 _amountToSwap, uint256 _minTokenReceived, bool _swapWETH) external nonReentrant onlyGovernorOrStrategist {
+        _rebalance(_amountToSwap, _minTokenReceived, _swapWETH);
+    }
+
+    /// @notice Only used for the initial deposit (or after calling withdrawAll) when there is no liquidity
+    ///         in the [-1, 0] tick. This function is needed since we can not swap to the desired position within
+    ///         the pool if there is no liquidity in the [-1, 0] ticker
+    function depositLiquidity() external nonReentrant onlyGovernorOrStrategist {
+        require(tokenId == 0, "Liquidity already deposited");
+        _addLiquidity();
+    }
+
+    /**
+     * @dev Rebalance the pool to the desired token split
+     */
+    function _rebalance(uint256 _amountToSwap, uint256 _minTokenReceived, bool _swapWETH) internal {
+        _removePartialLiquidity(withdrawLiquidityShare);
+        _swapToDesiredPosition(_amountToSwap, _minTokenReceived, _swapWETH);
+        // calling check liquidity early so we don't get unexpected errors when adding liquidity
+        // in the later stages of this function
+        _checkLiquidityWithinExpectedShare();
+        _addLiquidity();
+        _checkLiquidityWithinExpectedShare();
+    }
+
+    /**
+     * @dev Decrease partial liquidity from the pool.
+     * @param _partialLiquidityToDecrease The amount of liquidity to remove expressed in 18 decimals
+     */
+    function _removePartialLiquidity(uint256 _partialLiquidityToDecrease) internal {
+        require(_partialLiquidityToDecrease > 0, "Must remove some liquidity");
+        require(_partialLiquidityToDecrease < 1e18, "Mustn't remove all liquidity");
+
+        _removeLiquidity(_partialLiquidityToDecrease);
+    }
+
+    /**
+     * @dev Remove all liquidity from the pool.
+     */
+    function _removeAllLiquidity() internal {
+        _removeLiquidity(1e18);
+
+        positionManager.burn(tokenId);
+        tokenId = 0;
```

Comment:
Throughout the codebase, the assumption is that token is assumed to be non-existent if it has the id `0`. However Aerodrome's NonfungiblePositionManager can also have a NFT with ID 0. It's impossible for the strategy to get the NFT with 0 ID though. So, it's fine. But just flagging it in case

Reply:
Yup have though about that as well. Thanks for pointing it out